source/codegen/templates/service_registrar.cpp.mako

```diff
@@ -29,15 +29,19 @@ namespace ${namespace} {
 namespace {
 struct LibraryAndService {
   LibraryAndService(
-    const ${resource_repository_type}& resource_repository,
+% for resource_repository_dep in resource_repository_deps:
+    const ${resource_repository_dep.resource_repository_type}& ${resource_repository_dep.local_name},
+% endfor
 % for cross_driver_dep in cross_driver_session_deps:
```

Comment:
Nit: it seems like a lot of these back-to-back session_deps for loops could be combined with just anew helper method that gets all of them. It's OK how it is though.

Reply:
I have merged lists for `cross_driver_session_deps` and `resource_repository_deps` using helper methods and replaced all these back to back loops with single loop.